docs/data/joy/components/accordion/AccordionTransition.js

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import Box from '@mui/joy/Box';
+import AccordionGroup from '@mui/joy/AccordionGroup';
+import Accordion from '@mui/joy/Accordion';
+import AccordionDetails from '@mui/joy/AccordionDetails';
+import AccordionSummary from '@mui/joy/AccordionSummary';
+import RadioGroup from '@mui/joy/RadioGroup';
+import Radio from '@mui/joy/Radio';
+import Stack from '@mui/joy/Stack';
+import Typography from '@mui/joy/Typography';
+import HighlightedCode from 'docs/src/modules/components/HighlightedCode';
+import BrandingProvider from 'docs/src/BrandingProvider';
+
+export default function AccordionTransition() {
+  const [transition, setTransition] = React.useState('0.4s ease');
+  return (
+    <Stack alignItems="center" spacing={2} sx={{ flex: 1 }}>
+      <RadioGroup
+        orientation="horizontal"
+        value={transition}
+        onChange={(event) => {
+          setTransition(event.target.value);
+        }}
+      >
+        <Radio value="0.3s linear" label="Linear" />
```

Comment:
This easing makes me feel uncomfortable. What if instead, we were slowing the default animation? Or have it to be instant?

Reply:
I think the default animation/transition is out of scope from this PR that's why the demo illustrates how to add transitions. 

@zanivan mentioned the default transition for Joy as well and I think it should be done separately so that each component does not need to wait to roll out.

Reply:
I think adding default animations/transitions to Joy UI would be really helpful. It could improve the microinteractions on components and potentially become a standout feature that sets Joy UI apart from competitors. Also, if we incorporate the Joy UI design language into Base UI, it would further differentiate the two products.

All in all, I think we can polish out a bit these animations, the easing is indeed a bit off, mostly on the bouncing demo—at least for me. Are there any workarounds to improve these animations before coming up with the default transition?

Reply:
> I think the default animation/transition is out of scope from this PR that's why the demo illustrates how to add transitions.

@siriwatknp Ah sorry, I used the wrong terminology. By "default animation", I meant the one applied on the demo.

But actually, this connects well to point 1. in https://github.com/mui/material-ui/pull/38164#pullrequestreview-1549831929. This feels OK, maybe a 10% faster would be nice

https://github.com/mui/material-ui/assets/3165635/1c145b3a-66d5-4970-837f-c3bfb0e38335

This feels too fast; 40% slower would be great:

https://github.com/mui/material-ui/assets/3165635/4073ac39-6e8e-4899-9d54-3d78f9f56100

Reply:
I think we can also see another bug in this animation, the children isn't stable, "The Accordion Group supports the…" moves, so to start reading the content, you have to wait for the whole transition to finish

https://github.com/mui/material-ui/assets/3165635/e6f91ae3-2287-43ad-b742-22e1c505eb77

It's better in https://ui.shadcn.com/docs/components/accordion.